lib/Provider/Command/MessageSend.php

```diff
@@ -0,0 +1,137 @@
+		} catch (\Throwable $e) {
+			throw new SendException('Error: occurred while retrieving mail account details', 0, $e);
+		}
+		// convert mail provider message to mail app message
+		$localMessage = new LocalMessage();
+		$localMessage->setType($localMessage::TYPE_OUTGOING);
+		$localMessage->setAccountId($account->getId());
+		$localMessage->setSubject($message->getSubject());
+		$localMessage->setBody($message->getBody());
+		// disabled due to issues caused by opening these messages in gui
+		//$localMessage->setEditorBody($message->getBody());
+		$localMessage->setHtml(true);
+		$localMessage->setSendAt($this->time->getTime());
+		// convert all mail provider attachments to local attachments
+		$attachments = [];
+		if (count($message->getAttachments()) > 0) {
+			// iterate attachments and save them
+			foreach ($message->getAttachments() as $entry) {
+				// determine if required parameters are set
+				if (empty($entry->getName()) || empty($entry->getType()) || empty($entry->getContents())) {
+					throw new SendException('Invalid Attachment Parameter: MUST contain values for Name, Type and Contents');
+				}
+				// convert mail provider attachment to mail app attachment
+				try {
+					$attachments[] = $this->attachmentService->addFileFromString(
+						$userId,
+						$entry->getName(),
+						$entry->getType(),
+						$entry->getContents()
+					)->jsonSerialize();
+				} catch (\Throwable $e) {
+					throw new SendException('Error: occurred while saving mail message attachment', 0, $e);
+				}
+			}
+		}
+		// determine if required To address is set
+		if (empty($message->getTo()) || empty($message->getTo()[0]->getAddress())) {
+			throw new SendException('Invalid Message Parameter: MUST contain at least one TO address with a valid address');
+		}
+		// convert recipient addresses
+		$to = $this->convertAddressArray($message->getTo());
+		$cc = $this->convertAddressArray($message->getCc());
+		$bcc = $this->convertAddressArray($message->getBcc());
+		// save message for sending
+		try {
+			$localMessage = $this->outboxService->saveMessage(
+				$account,
+				$localMessage,
+				$to,
+				$cc,
+				$bcc,
+				$attachments
+			);
+		} catch (\Throwable $e) {
+			throw new SendException('Error: occurred while saving mail message', 0, $e);
+		}
+		// evaluate if system messages are to be sent instantly
+		if ($this->config->getAppValue(Application::APP_ID, 'send_system_messages_instantly', 'yes') === 'yes') {
+			try {
+				$localMessage = $this->outboxService->sendMessage($localMessage, $account);
+			} catch (\Throwable $e) {
+				throw new SendException('Error: occurred while sending mail message', 0, $e);
+			}
+		}
+
+		return $localMessage;
+	}
+
+	/**
+	 * Converts IAddress objects collection to plain array
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param array<int,IAddress> $addresses collection of IAddress objects
+	 *
+	 * @return array<int,array{email: string, label?: string}>
```

Comment:
```suggestion
	 * @return array<int,array{email: ?string, label?: ?string}>
```

Reply:
This didn't work, caused more Psalm errors

Reply:
```
ERROR: InvalidArgument - lib/Provider/Command/MessageSend.php:109:5 - Argument 3 of OCA\Mail\Service\OutboxService::saveMessage expects array<int, array<array-key, string>>, but array<int, array{email: null|string, label?: string}> provided (see https://psalm.dev/004)
				$to,

ERROR: InvalidArgument - lib/Provider/Command/MessageSend.php:110:5 - Argument 4 of OCA\Mail\Service\OutboxService::saveMessage expects array<int, array<array-key, string>>, but array<int, array{email: null|string, label?: string}> provided (see https://psalm.dev/004)
				$cc,

ERROR: InvalidArgument - lib/Provider/Command/MessageSend.php:111:5 - Argument 5 of OCA\Mail\Service\OutboxService::saveMessage expects array<int, array<array-key, string>>, but array<int, array{email: null|string, label?: string}> provided (see https://psalm.dev/004)
				$bcc,
```

https://github.com/nextcloud/mail/pull/10114

Reply:
```php
	/**
	 * Converts IAddress objects collection to plain array
	 *
	 * @since 4.0.0
	 *
	 * @param array<int,IAddress> $addresses collection of IAddress objects
	 *
	 * @return array<int,array{email: ?string, label?: string}>
	 */
	protected function convertAddressArray(array $addresses): array {
		return array_map(static function (IAddress $address) {
			$label = $address->getLabel();
			return !empty($label)
				? ['email' => $address->getAddress(), 'label' => $label]
				: ['email' => $address->getAddress()];
		}, $addresses);
	}
```

Please update the function as above.

- The annotated return type for label is always a string.
- It's required to introduce a local variable for label, otherwise psalm is unable to detect.